### PR TITLE
added variants to buttonIcon

### PR DESCRIPTION
**Pull request by oscarotero**
ButtonIcon component have now two variants: `normal` and `destructive`, to display the icon in gray (by default) or red.

### CODE REVIEW

**Comment by pedroparra** [comment]
Hi @oscarotero.
I think this variant should be a property of the icon, not of the buttonIcon.
ButtonGroup doesn't have destructive variants, but Button does.
Alert icon can be followed by a danger text in a section, or button, and if variant desctructive only can be used with buttonIcon,  we will be forced to put a class for show the desctrucitve icon.

File: src/ButtonIcon/index.js [hunk 1]
```diff
@@ -3,18 +3,27 @@ import {default as PT} from 'prop-types'
...
+    {[`is-${variant}`]: variant },
+    extraClass
+  )
+  return (<ElementType {...props} className={styles}>{children}</ElementType>)
 }
 
 ButtonIcon.defaultProps = {
-  as: 'button'
+  as: 'button',
+  variant: null
```
Comment (pedroparra) [inline src/ButtonIcon/index.js]: variant cant be null in this case. PropTypes force it to be one of ['normal' or 'desctructive'] by default.

Comment (oscarotero) [reply src/ButtonIcon/index.js]: I just replicate how variants are in Button (that is null by default). But I can change to `normal` by default.

**Comment by oscarotero** [comment]
@pedroparra The variant in buttonIcon does not affect only to the color of icon, but also different states (like hover, focus, etc).
We could add variants to icons too, (I'm not sure this is a good idea, we have the class helpers for that (example: `<Icon name="warning" extraClass="ola-error" />`), but this shouldn't prevent to have variants in ButtonIcon anyway.

File: src/ButtonIcon/index.js (hunk 1 above)
Comment (pedroparra) [reply src/ButtonIcon/index.js]: We must fix the null variants if they do not have a prop type null ...

**Comment by pedroparra** [comment]
@oscarotero We can try.
I'm afraid to see it all full of helper ( extraClass="ola-error" ) ;)

**Comment by oscarotero** [comment]
Added also the `chevron` variant to fix #89

**Pull request merged**